build.gradle

```diff
@@ -191,3 +175,22 @@ publishing {
 test {
     maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1
 }
+
+def readVersion() {

```

Comment:
I would rewrite it a little, but adding the `version == 'unspecified'` check to the readVersion function totally makes sense.
```
def readVersion() {
    if (version == 'unspecified') {
        Properties properties = new Properties()
        new File('version.properties').withInputStream {
            properties.load(it)
        }
        if (properties.version != null) {
            return properties.version
        }
    }
    return version
}
```

Reply:
my approach was to only have one "return" in the function, but i can fix this ;-)